### PR TITLE
Added some new documentation

### PR DESCRIPTION
**Pull request by lintusj1**

### CODE REVIEW

File: docs/good-to-know.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Good to know
```
Comment (vuolleko) [inline docs/good-to-know.rst]: This is super good to have! :)

File: docs/good-to-know.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Here we describe some important concepts related to ELFI. These will help in understanding
+how to implement custom operations (such as simulators or summaries) and can potentially
+save the user from some pitfalls.
+
+Generative model
+----------------
+
+By a generative model we mean any model that can generate some data. In ELFI the
+generative model is described with a directed acyclic graph (DAG) and the representation
+is stored in the `ElfiModel` instance. In LFI it typically includes everything from prior
```
Comment (vuolleko) [inline docs/good-to-know.rst]: Perhaps in this context don't use 'LFI' or better yet define it in a separate entry. 

Could just point to Wikipedia for some words? Also for DAG.

File: docs/developer/architecture.rst [hunk 3]
```diff
@@ -1,2 +1,98 @@
 ELFI architecture
-=================
\ No newline at end of file
+=================
+
+Here we explain the internal representation of the generative model in ELFI. This
+representation contains everything that is needed to generate data, but is separate from
+e.g. the inference methods or the data storages. This information is aimed for developers
+and is not essential for using ELFI. We assume the reader is quite familiar with Python
+and has perhaps already read some of its source code.
```
Comment (vuolleko) [inline docs/developer/architecture.rst]: Source code of Python?

File: docs/developer/architecture.rst [hunk 4]
```diff
@@ -1,2 +1,98 @@
...
+In the standard compilation process, the `source_net` is augmented with additional nodes
+such as batch_size or random_state, that are then added as dependencies for those
+operations that require them. In addition the state dicts will be turned into either a
+runnable operation or a precomputed value.
+
+The execution order of the nodes in the compiled graph follows the topological ordering of
+the DAG (dependency order) and is guaranteed to be the same every time. Note that because
+the default behaviour is that nodes share a random state, changing a node that uses shared
+random state will affect the result of any later node in the ordering using the same
+shared random state even if they would not be depended based on the graph topology. If
```
Comment (vuolleko) [inline docs/developer/architecture.rst]: Typo: dependent

Better to say independent?

File: docs/developer/architecture.rst (hunk 3 above)
Comment (lintusj1) [inline docs/developer/architecture.rst]: Hehe, true :D I'll fix it :+1:

**Pull request merged**